### PR TITLE
feat(core): add long arithmetic functions

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
I didn't export the precision function because it only returns what is useful to the other daff arithmetic functions. I think someone might expect it to return 4 instead of 10000.

### CODE REVIEW

File: libs/core/src/utils/long-arithmetic.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		(acc, number) => acc / Math.round(number*precision), 
+		Math.round(numbers[0]*precision)
+	) * Math.pow(precision, numbers.length - 2);
+}
+
+/**
+ * A helper function to get the number of decimal significant figures of a number.
+ * @param number
+ */
+function daffPrecision(number: number) {
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/core/src/utils/long-arithmetic.ts:56]: I could just update it to use BigInt. Then we wouldn't need the doc

[the remark this answers is not in the document]
Comment (griest024) [reply libs/core/src/utils/long-arithmetic.ts:56]: I think we need polyfills for IE. If its easy I would go ahead and do it though.

[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/core/src/utils/long-arithmetic.ts:70]: I looked at BigInt and I don't think we can actually use it for this. The problem is that we need to divide at the end, and dividing means decimals. And decimals in BigInt are just floored. I added additional docs for each function explaining what their limitations are.

[the remark this answers is not in the document]
Comment (griest024) [reply libs/core/src/utils/long-arithmetic.ts:70]: hm well I think the only think that would need to be a `BigInt` is `number * p` and maybe `p` itself? I'm not sure and I don't think its worth worrying about it now.

**Pull request merged**